src/coreclr/vm/methodtablebuilder.cpp

```diff
@@ -1493,13 +1493,8 @@ MethodTableBuilder::BuildMethodTableThrowing(
             IfFailThrow(GetMDImport()->GetNameOfTypeDef(bmtInternal->pType->GetEnclosingTypeToken(), NULL, &nameSpace));
         }
 
-#if defined(TARGET_ARM64)
-        // All the funtions in System.Runtime.Intrinsics.Arm are hardware intrinsics.
-        if (hr == S_OK && strcmp(nameSpace, "System.Runtime.Intrinsics.Arm") == 0)
-#else
-        // All the funtions in System.Runtime.Intrinsics.X86 are hardware intrinsics.
-        if (hr == S_OK && (strcmp(nameSpace, "System.Runtime.Intrinsics.X86") == 0))
-#endif
+        // All the funtions in System.Runtime.Intrinsics are potentially hardware intrinsics.
+        if (hr == S_OK && strncmp(nameSpace, "System.Runtime.Intrinsics", 25) == 0)
```

Comment:
nit
```c
const char* prefix = "System.Runtime.Intrinsics";
if (hr == S_OK && strncmp(nameSpace, prefix, strlen(prefix)) == 0)
```

Reply:
I don't believe we currently follow a similar pattern elsewhere.

There are also considerations that we have many wrappers for various CRT functions in the PAL layers and so `strlen` might not be constant folded as intended.